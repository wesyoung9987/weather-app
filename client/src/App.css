* {
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
  box-sizing: border-box;
}

.App .router-container {
  margin-top: 110px;
}
