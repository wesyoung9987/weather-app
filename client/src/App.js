import React from 'react';
import {
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';

import Header from './components/Header';
import Cities from './components/Cities';
import Weather from './components/Weather';
import './App.css';

function App() {
  return (
    <div className="App">
      <Header />

      <div className="router-container">
        <Switch>
          <Route path="/weather">
            <Weather />
          </Route>
          <Route path="/">
            <Cities />
          </Route>
          <Redirect to='/' />
        </Switch>
      </div>
    </div>
  );
}

export default App;
