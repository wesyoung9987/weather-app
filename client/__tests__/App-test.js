import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

jest.mock('react-router-dom', () => {
  return {
    Link: ({ children, to }) => {
      return (
        <a href={to}>{children}</a>
      );
    },
    Switch: ({ children }) => {
      return (
        <div>{children}</div>
      );
    },
    Route: ({ children }) => {
      return (
        <div>{children}</div>
      );
    },
    Redirect: ({ children, to }) => {
      return (
        <div to={to}>{children}</div>
      );
    },
    useHistory: () => {
      return {
        push: jest.fn(),
      };
    },
    useLocation: () => {
      return {
        search: 'test=123',
      };
    },
  };
});

describe('App', () => {
  test('it should render correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

