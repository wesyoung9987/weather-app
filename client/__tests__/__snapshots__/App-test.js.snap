// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it should render correctly 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <div
      className="Search"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-container"
        >
          <label
            htmlFor="zip-code-input"
          >
            Search Weather by Zip Code:
          </label>
          <input
            id="zip-code-input"
            onChange={[Function]}
            placeholder="Enter zip code..."
            type="text"
            value=""
          />
        </div>
        <div
          className="button-container"
        >
          <button
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="router-container"
  >
    <div>
      <div>
        <div
          className="Weather"
        >
          <div
            className="CurrentWeather"
          >
            <h1 />
            <div
              className="row"
            >
              <div
                className="group"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="Cities"
        >
          <div
            className="City"
          >
            <a
              href="/weather?zip=98101"
            >
              <div
                className="city-inner"
                style={
                  Object {
                    "backgroundImage": "url(https://images.unsplash.com/photo-1502175353174-a7a70e73b362?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8c2VhdHRsZXxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&w=1000&q=80)",
                  }
                }
              >
                <div
                  className="city-overlay"
                >
                  <p
                    className="city-name"
                  >
                    Seattle
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            className="City"
          >
            <a
              href="/weather?zip=29401"
            >
              <div
                className="city-inner"
                style={
                  Object {
                    "backgroundImage": "url(https://images.squarespace-cdn.com/content/v1/54e4e6bfe4b04465619a3b2f/1428181383386-4JQLXZTIPNQBFTGT8XM1/ke17ZwdGBToddI8pDm48kFvMGRYkbCNsuAMUea4LQlwUqsxRUqqbr1mOJYKfIPR7LoDQ9mXPOjoJoqy81S2I8N_N4V1vUb5AoIIIbLZhVYxCRW4BPu10St3TBAUQYVKccdYVXBVvju5Hk2C6Ed5VluIPS4Tsv7BEViXnheA8Xz_IC4Rz-jfCe7mDZQ8G-Nbc/Charleston+Daytime.jpg?format=2500w)",
                  }
                }
              >
                <div
                  className="city-overlay"
                >
                  <p
                    className="city-name"
                  >
                    Charleston
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            className="City"
          >
            <a
              href="/weather?zip=84101"
            >
              <div
                className="city-inner"
                style={
                  Object {
                    "backgroundImage": "url(https://dynamic-media-cdn.tripadvisor.com/media/photo-o/15/34/00/f0/salt-lake-city.jpg?w=1100&h=600&s=1)",
                  }
                }
              >
                <div
                  className="city-overlay"
                >
                  <p
                    className="city-name"
                  >
                    Salt Lake City
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            className="City"
          >
            <a
              href="/weather?zip=10001"
            >
              <div
                className="city-inner"
                style={
                  Object {
                    "backgroundImage": "url(https://dynaimage.cdn.cnn.com/cnn/c_fill,g_auto,w_1200,h_675,ar_16:9/https%3A%2F%2Fcdn.cnn.com%2Fcnnnext%2Fdam%2Fassets%2F210409114908-nyc-april-2021-tease.jpg)",
                  }
                }
              >
                <div
                  className="city-overlay"
                >
                  <p
                    className="city-name"
                  >
                    New York City
                  </p>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        to="/"
      />
    </div>
  </div>
</div>
`;
